wal/wal.go

```diff
@@ -330,27 +330,33 @@ func (w *WAL) sync() error {
 	return err
 }
 
-// ReleaseLockTo releases the locks w is holding, which
-// have index smaller or equal to the given index.
+// ReleaseLockTo releases the locks w is holding so that
+// w only holds one lock that has an index smaller than
+// the given index.
 func (w *WAL) ReleaseLockTo(index uint64) error {
-	for _, l := range w.locks {
-		_, i, err := parseWalName(path.Base(l.Name()))
-		if err != nil {
-			return err
-		}
-		if i > index {
-			return nil
-		}
-		err = l.Unlock()
+	var smaller int
+
+	for i, l := range w.locks {
+		_, lockIndex, err := parseWalName(path.Base(l.Name()))
 		if err != nil {
 			return err
 		}
-		err = l.Destroy()
-		if err != nil {
-			return err
+		if lockIndex > index {
+			smaller = i - 1
+			break
 		}
-		w.locks = w.locks[1:]
 	}
+
+	if smaller <= 0 {
+		return nil
+	}
+
+	for i := 0; i < smaller; i++ {
+		w.locks[i].Unlock()
+		w.locks[i].Destroy()
+	}
+	w.locks = w.locks[smaller-1:]
```

Comment:
should this be `[smaller:]` because we just want the file at the `smaller` index?